### ISSUE TITLE
Fix regex in assert_sepa_id_set* functions

### ISSUE THREAD

**Issue by kewisch** [kewisch/sepa.js#28]
https://github.com/kewisch/sepa.js/blob/40f2ee0255d5907408c530f292c069d8904fed8e/lib/sepa.js#L796

The regex matches any characters in the string, and is also put together wrong (the pipe sign is not "or", it is a character match). I don't know how I managed to produce that.

**Comment by lippertto** [comment]
Hi, I am looking at this bug. Writing a new regex should be easy enough.

I have one remark though: From what I understand, banks may bilaterally agree on using an extended character set, e.g., in Greece. If we change this behavior here, it will no longer be possible to create documents for this case. But I do not have a lot experience with SEPA and do not even know where to find authoritative information.

Do you have any thoughts on this?

If we keep the assertions, I would like to rename the functions `assert_sepa_id_set1` and `assert_sepa_id_set2` to `assert_extended_latin_characterset` and `assert_extended_latin_characterset_with_space` to make clearer what they do.

**Comment by kewisch** [comment]
Ah good question. I don't have enough hands on SEPA experience on this either unfortunately. Is this documented in the spec somewhere? If not, then I'd say we just go with what the spec says, and wait until someone files a bug with a more specific scenario.

I believe I used set1 and set2 since those sets were defined as such in the spec. I guess maybe `assert_sepa_set1_latin_extended_with_space` and `assert_sepa_set2_latin_extended` ?

**Comment by lippertto** [comment]
What I found on the internet by [European Payments Council](https://www.europeanpaymentscouncil.eu/document-library/guidance-documents/sepa-requirements-extended-character-set-unicode-subset-best):
> Within the SEPA schemes, unless otherwise agreed, only a limited set of Latin characters may be exchanged. This restriction was agreed in order to overcome the complexity that would result if all European language based characters were to be exchanged between all adhering banks.
However by prior agreement banks may exchange additional characters, perhaps through bilateral/multilateral agreements or via an agreed AOS (Additional Optional Service).

My take away is that it is possible to implement AOS for more additional characters.

Another thing that I found by a German company that does consultancy on SEPA ([Hettwer GmbH](https://www.hettwer-beratung.de/sepa-spezialwissen/sepa-technische-anforderungen/sepa-verwendungszweck/))
> To date, there are no plans to support the national character set (umlauts) as part of an AOS for national SEPA payments, as is the case in Greece.

I take from it that an AOS is in place for Greece.

Also, I found a [list of AOS](https://www.europeanpaymentscouncil.eu/what-we-do/epc-payment-schemes/sepa-credit-transfer/sepa-credit-transfer-additional-optional) . Apparently Finland also allows additional characters.


Altogether, I would suggest to remove the assertions and only check for the length (i.e. 35) for both.
